crypto/pem/pvkfmt.c

```diff
@@ -1078,33 +1079,28 @@ static int i2b_PVK(unsigned char **out, const EVP_PKEY *pk, int enclevel,
 #endif
     }
 
-    EVP_CIPHER_CTX_free(cctx);
-#ifndef OPENSSL_NO_RC4
-    EVP_CIPHER_free(rc4);
-#endif
     if (*out == NULL)
         *out = start;
-
-    return outlen;
-
+    ret = outlen;
  error:
     EVP_CIPHER_CTX_free(cctx);
 #ifndef OPENSSL_NO_RC4
     EVP_CIPHER_free(rc4);
 #endif
     if (*out == NULL)
         OPENSSL_free(start);
```

Comment:
This OPENSSL_free(start) is not so good on success

Reply:
On success *out should not be NULL, so start should be safe.